### PR TITLE
Anonymize paths in error messages

### PR DESCRIPTION
**Pull request by fvictorio**

### CODE REVIEW

File: packages/buidler-core/src/internal/sentry/anonymizer.ts [hunk 1]
```diff
@@ -91,6 +91,12 @@ export class Anonymizer {
     };
   }
 
+  public anonymizeErrorMessage(errorMessage: string): string {
+    // the \\ before path.sep is necessary for this to work on windows
+    const pathRegex = new RegExp(`\\S+\\${path.sep}\\S+`, "g");
+    return errorMessage.replace(pathRegex, "<user-file>");
```
Comment (alcuadrado) [inline packages/buidler-core/src/internal/sentry/anonymizer.ts:97]: Shouldn't this be replaceAll?

File: packages/buidler-core/test/internal/sentry/anonymizer.ts [hunk 2]
```diff
@@ -118,4 +118,83 @@ describe("Anonymizer", () => {
       assert.isTrue(anonymizationResult.anonymizeContent);
     });
   });
+
```
Comment (alcuadrado) [inline packages/buidler-core/test/internal/sentry/anonymizer.ts:121]: I think this tests won't pass on windows

File: packages/buidler-core/src/internal/sentry/anonymizer.ts (hunk 1 above)
Comment (fvictorio) [reply packages/buidler-core/src/internal/sentry/anonymizer.ts:97]: The regex has the `g` modifier. Plus, `replaceAll` was added to V8 like, two weeks ago? :sweat_smile:

File: packages/buidler-core/test/internal/sentry/anonymizer.ts (hunk 2 above)
Comment (fvictorio) [reply packages/buidler-core/test/internal/sentry/anonymizer.ts:121]: They seem to have passed

File: packages/buidler-core/src/internal/sentry/anonymizer.ts (hunk 1 above)
Comment (alcuadrado) [reply packages/buidler-core/src/internal/sentry/anonymizer.ts:97]: Haha I didn't notice that when I Googled if it existed

File: packages/buidler-core/test/internal/sentry/anonymizer.ts (hunk 2 above)
Comment (alcuadrado) [reply packages/buidler-core/test/internal/sentry/anonymizer.ts:121]: I'm confused. Path.sep shouldn't match slash there. Right?

Comment (alcuadrado) [reply packages/buidler-core/test/internal/sentry/anonymizer.ts:121]: Nevermind, you are constructing test paths using things like __filename

**Pull request merged**